File: Frontend/barbershop/src/types/ResponseDTOs/branchesProductDTO.ts
// Make sure to import related DTOs
import type { BranchDTO } from './branchDTO';
import type { ProductDTO } from './productDTO';
import type { SizeDTO } from './sizeDTO';

export type BranchesProductDTO = {
    branchesProductsId: number;
    branchId?: number;
    productId?: number;
    quantity?: number;
    sizeId?: number;
    updateAt?: Date;
    branchDTO?: BranchDTO;
    productDTO?: ProductDTO;
    sizeDTO?: SizeDTO;
};

File: Backend/barbershop/barbershop/Models/ResponseDTOs/BranchesProductDTO.cs
using barbershop.Models.Entitys;

namespace barbershop.Models.ResponseDTOs
{
    public class BranchesProductDTO
    {
        public int BranchesProductsId { get; set; }

        public int? BranchId { get; set; }

        public int? ProductId { get; set; }

        public int? Quantity { get; set; }

        public int? SizeId { get; set; }
        public bool? IsActive { get; set; }

        public DateTime? UpdateAt { get; set; }

        public  BranchDTO? BranchDTO { get; set; }

        public  ProductDTO? ProductDTO { get; set; }

        public  SizeDTO? SizeDTO { get; set; }

    }
}
